### ISSUE TITLE
Trinity raises `AlreadyWaiting` error semi-regularly

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/py-evm#737]
### What is wrong?

Encountered when running in full/fast sync mode.

```bash
   ERROR  05-20 15:16:13  base_events  Task exception was never retrieved
future: <Task finished coro=<PeerPool.lookup_random_node() done, defined at /Users/piper/sites/py-evm/p2p/peer.py:719> exception=AlreadyWaiting("There's another coroutine waiting for a ping packet from <Node(0x47ca@165.227.61.121)>",)>
Traceback (most recent call last):
  File "/Users/piper/sites/py-evm/p2p/peer.py", line 723, in lookup_random_node
    await self.discovery.lookup_random(self.cancel_token)
  File "/Users/piper/sites/py-evm/p2p/discovery.py", line 90, in lookup_random
    return await self.kademlia.lookup_random(self.cancel_token.chain(cancel_token))
  File "/Users/piper/sites/py-evm/p2p/kademlia.py", line 575, in lookup_random
    return await self.lookup(random.randint(0, k_max_node_id), cancel_token)
  File "/Users/piper/sites/py-evm/p2p/kademlia.py", line 565, in lookup
    *[_find_node(node_id, n) for n in nodes_to_ask])
  File "/Users/piper/sites/py-evm/p2p/kademlia.py", line 550, in _find_node
    bonded = await asyncio.gather(*[self.bond(c, cancel_token) for c in candidates])
  File "/Users/piper/sites/py-evm/p2p/kademlia.py", line 514, in bond
    await self.wait_ping(node, cancel_token)
  File "/Users/piper/sites/py-evm/p2p/kademlia.py", line 413, in wait_ping
    "There's another coroutine waiting for a ping packet from {}".format(remote))
p2p.kademlia.AlreadyWaiting: There's another coroutine waiting for a ping packet from <Node(0x47ca@165.227.61.121)>
```

This exception happens semi-regularly when syncing.

**Comment by gsalgado** [comment]
This is something that needs fixing, for sure, but it should only happen if we run a standard `PeerPool`, which uses discovery to find nodes, but that's probably not a good idea as in my tests the discovery v4 never seems to work at finding LES peers. And actually, trinity seems to use a `HardcodedNodesPeerPool` for both light/full modes, so I'm guessing you're running a locally-modified version?

**Comment by pipermerriam** [comment]
Yes, I am running with `PeerPool`

**Comment by pipermerriam** [comment]
Fixed by #765
